ttnn/cpp/ttnn/operations/data_movement/reshape_view/device/host/reshape_rm_host_prep.cpp

```diff
@@ -0,0 +1,216 @@
+    bool dest_page_is_pow_2 = tt::tt_metal::is_power_of_two_at_least_32(dest_page_size_bytes);
+    uint32_t dest_page_pow_2 = dest_page_is_pow_2 ? (std::uint32_t)std::log2(dest_page_size_bytes) : 0;
+    std::vector<uint32_t> compile_time_args = {
+        (std::uint32_t)src0_is_dram,
+        (std::uint32_t)(source_page_size_bytes % 64 == 0) ? 1 : 0,
+        (std::uint32_t)(source_page_size_bytes % 16 == 0) ? 1 : 0,
+        src0_cb_index,
+        src1_cb_index,
+        source_page_size_bytes,
+        dest_page_size_bytes,
+        source_page_is_pow_2,
+        source_page_pow_2,
+        dest_page_is_pow_2,
+        dest_page_pow_2};
+
+    tt::tt_metal::KernelHandle reader_kernel_id = tt::tt_metal::CreateKernel(
+        program,
+        "ttnn/cpp/ttnn/operations/data_movement/reshape_view/device/device/rm_reshape_interleaved.cpp",
+        total_cores,
+        tt::tt_metal::ReaderDataMovementConfig(compile_time_args));
+    uint32_t done = 0;
+    for (int core_x = 0; core_x < num_cores_x; core_x++) {
+        for (int core_y = 0; core_y < num_cores_y; core_y++) {
+            CoreCoord core = {core_x, core_y};
+            if (done == 1) {
+                const std::vector<uint32_t> reader_runtime_args = {
+                    src_buffer->address(), dst_buffer->address(), source_read_size_bytes, 0, 0, 0, 0, 1
+
+                };
+                tt::tt_metal::SetRuntimeArgs(program, reader_kernel_id, core, reader_runtime_args);
+            } else {
+                // Create the circular buffers
+
+                // set the runtime args
+                // set the compile time args
+                const uint32_t start_of_read = read_start_page;
+                uint32_t end_of_read = read_start_page + responsibility;
+                end_of_read = end_of_read < input_log_shape[-2] ? end_of_read : input_log_shape[-2];
+
+                const std::vector<uint32_t> reader_runtime_args = {
+                    src_buffer->address(),
+                    dst_buffer->address(),
+                    source_read_size_bytes,
+                    start_of_read,
+                    end_of_read,
+                    write_start_page,
+                    write_start_offset,
+                    done
+
+                };
+                write_start_offset += offset_jump;
+                if (write_start_offset >= dest_page_size_bytes) {
+                    write_start_page += write_jump + 1;
+                    write_start_offset -= dest_page_size_bytes;
+                } else {
+                    write_start_page += write_jump;
+                }
+                read_start_page = end_of_read;
+                done = (end_of_read == input_log_shape[-2]) ? 1 : 0;
+                tt::tt_metal::SetRuntimeArgs(program, reader_kernel_id, core, reader_runtime_args);
+            }
+        }
+    }
+    auto override_runtime_args_callback = [reader_kernel_id, compute_with_storage_grid_size](
```

Comment:
move to a separate method?

Reply:
Almost every OP does it this way though, I wanted to maintain consistency.

Reply:
That style is deprecated now. For existing ops it's fine, but we do prefer this variant:

https://docs.tenstorrent.com/tt-metal/latest/ttnn/ttnn/adding_new_ttnn_operation.html